pkg/storage/db.go

```diff
@@ -120,6 +122,61 @@ func (db *DB) Init(ctx context.Context, postgresConfig *PostgresConfig) error {
 	return initerr
 }
 
+// GeneratePartition finds the latest partition, and generate the next partition based on the previous partions time range
+func (db *DB) GeneratePartition(ctx context.Context) error {
+	stmt := `SELECT tablename FROM pg_catalog.pg_tables WHERE tablename LIKE 'aws_events_ips_hostnames_%' ORDER BY tablename DESC LIMIT 1`
```

Comment:
Suggestion: This is a clever way to ensure the always fast-forward partitioning but you're paying a premium in code logic and table naming contracts for it to work. You might consider using the same approach as a lot of DB schema migration tools and create a table that tracks the metadata you need to manage the schema. For example, you might have:

```sql
CREATE TABLE partitions (
  ts timestamp
  partition_begin timestamp
  partition_duration interval
)
```
And then use it with something like:

```sql
SELECT * FROM partitions ORDER BY partition_begin DESC LIMIT 1
```
Which would give you back pre-structured data for all the time values you need to generate the next partition. You'd insert a record into the table in the same transaction as creating the new table.

You'd buy yourself a couple of benefits with this approach. One is that it will remove all of the table name string parsing code and replace it with native types. Another is that it would enable you to adjust the partition duration at any time via configuration because the durations of each partition are recorded in the metadata rather than assumed.